roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-dimension.conf.j2

```diff
@@ -65,9 +65,27 @@ server {
 
 	ssl_certificate {{ matrix_ssl_config_dir_path }}/live/{{ matrix_nginx_proxy_proxy_dimension_hostname }}/fullchain.pem;
 	ssl_certificate_key {{ matrix_ssl_config_dir_path }}/live/{{ matrix_nginx_proxy_proxy_dimension_hostname }}/privkey.pem;
+
+	{% if matrix_nginx_proxy_ssl_config == "Modern" %}
```

Comment:
Assuming you adopt the other proposed changes, this can get simplified down to:

```nginx
ssl_protocols: {{ matrix_nginx_proxy_ssl_protocols }};

{% if matrix_nginx_proxy_ssl_ciphers == '' %}
ssl_prefer_server_ciphers: no;
{% else %}
ssl_prefer_server_ciphers: yes;
ssl_ciphers: {{ matrix_nginx_proxy_ssl_ciphers }};
{% endif %}
```